backend/report.js

```diff
@@ -294,6 +304,11 @@ function createProjection(filterObj) {
     if(filterObj.isCommentNeeded) {
         projection.comments = '$comments';
     }
+    if(filterObj.isCommentNeeded) {
```

Comment:
Why do you use the same condition "isCommentNeeded"? 
filterObj.isreadyForApproveNeeded ???